rules/consistent-indexof-check.js

```diff
@@ -0,0 +1,133 @@
+'use strict';
+const evaluateLiteralUnaryExpression = require('./utils/evaluate-literal-unaryexpression.js');
+const resolveVariableName = require('./utils/resolve-variable-name.js');
+const {isMethodCall} = require('./ast/index.js');
+
+const MESSAGE_ID = 'consistent-indexof-check';
+const messages = {
+	[MESSAGE_ID]: 'Prefer `{{replacement}}` over `{{value}}`.',
+};
+
+const comparisonMap = {
+	'<': '>',
+	'<=': '>=',
+	'>': '<',
+	'>=': '<=',
+};
+
+/**
+Determine the appropriate replacement based on the operator and value.
+
+@param {string} operator
+@param {number} value
+@param {string} variableName
+@returns {string | undefined}
+*/
+function getReplacement(operator, value, variableName) {
+	if ((operator === '<' && value <= 0) || (operator === '<=' && value <= -1)) {
+		return `${variableName} === -1`;
+	}
+
+	if ((operator === '>' && value === -1) || (operator === '>=' && value === 0)) {
+		return `${variableName} !== -1`;
+	}
+}
+
+/**
+Check if the node is a number literal or a unary expression resolving to a number.
+
+@param {import('estree').Node} node
+@returns {node is import('estree').Literal}
+*/
+function isNumberLiteral(node) {
+	if (node.type === 'UnaryExpression' && ['-', '+', '~'].includes(node.operator)) {
+		return isNumberLiteral(node.argument);
+	}
+
+	return node.type === 'Literal' && typeof node.value === 'number';
+}
+
+/**
+Find and process references to a given identifier.
+
+@param {import('eslint').Rule.RuleContext} context
+@param {import('estree').Identifier} identifierNode
+@returns
+*/
+function findAndProcessReferences(context, identifierNode) {
+	const variable = resolveVariableName(identifierNode.name, context.sourceCode.getScope(identifierNode));
```

Comment:
`sourceCode.getDeclaredVariables` should work here.

Reply:
Update in 4d88f256840842b5946add5b903e0f92f52f5bbf